sim/pokemon.ts

```diff
@@ -1110,20 +1110,29 @@ export class Pokemon {
 		return boosts;
 	}
 
+	getCappedBoost(boosts: SparseBoostsTable) {
+		const cappedBoost: SparseBoostsTable = {};
+		let boostName: BoostID;
+		for (boostName in boosts) {
+			const boost = boosts[boostName];
+			if (!boost) continue;
+			if (boost > 0) {
+				cappedBoost[boostName] = this.boosts[boostName] + boost > 6 ? 6 - this.boosts[boostName] : boost;
+			}
+			if (boost < 0) {
+				cappedBoost[boostName] = this.boosts[boostName] + boost < -6 ? -6 - this.boosts[boostName] : boost;
+			}
```

Comment:
Is this supposed to delete the boost if it was previously `0`? (I'm worried that you'll miss out on the "[POKEMON]'s [STAT] can't go any [DIRECTION]!" messages.) If not then I was thinking that `clampIntRange` might simplify this e.g.
```suggestion
			cappedBoost[boostName] = this.clampIntRange(this.boosts[boostName] + boost, -6, 6) - this.boosts[boostName];
```